x/ccv/provider/keeper/proposal.go

```diff
@@ -327,37 +347,54 @@ func (k Keeper) PendingStopProposalIterator(ctx sdk.Context) sdk.Iterator {
 // IteratePendingStopProposal iterates over the pending stop proposals in order and stop the chain if the stop time has passed,
 // otherwise it will break out of loop and return.
 func (k Keeper) IteratePendingStopProposal(ctx sdk.Context) {
+	propsToExecute := k.StopProposalsToExecute(ctx)
+
+	for _, prop := range propsToExecute {
+		err := k.StopConsumerChain(ctx, prop.ChainId, false, true)
+		if err != nil {
+			panic(fmt.Errorf("consumer chain failed to stop: %w", err))
+		}
+	}
+}
+
+// StopProposalsToExecute iterates over the pending stop proposals and returns an ordered list of stop proposals to be executed,
+// ie. consumer chains to stop. A prop is included in the returned list (and is deleted from store) if its proposed
+// stop time has passed.
+//
+// Note: this method is split out from IteratePendingCreateProposal to be easily unit tested.
+func (k Keeper) StopProposalsToExecute(ctx sdk.Context) []types.StopConsumerChainProposal {
+
+	// store the (to be) executed stop proposals in order
+	propsToExecute := []types.StopConsumerChainProposal{}
+
 	iterator := k.PendingStopProposalIterator(ctx)
 	defer iterator.Close()
 
 	if !iterator.Valid() {
-		return
+		return propsToExecute
 	}
 
-	// store the executed proposals in order
-	execProposals := []types.StopConsumerChainProposal{}
-
 	for ; iterator.Valid(); iterator.Next() {
+
 		key := iterator.Key()
 		stopTime, chainID, err := types.ParsePendingStopProposalKey(key)
 		if err != nil {
 			panic(fmt.Errorf("failed to parse pending stop proposal key: %w", err))
 		}
 
 		if !ctx.BlockTime().Before(stopTime) {
-			err = k.StopConsumerChain(ctx, chainID, false, true)
-			if err != nil {
-				panic(fmt.Errorf("consumer chain failed to stop: %w", err))
-			}
-			execProposals = append(execProposals,
+			propsToExecute = append(propsToExecute,
 				types.StopConsumerChainProposal{ChainId: chainID, StopTime: stopTime})
 		} else {
+			// No more proposals to check, since they're stored/ordered by timestamp.
 			break
 		}
 	}
 
-	// delete the proposals executed
-	k.DeletePendingStopProposals(ctx, execProposals...)
+	// delete the proposals to be executed
+	k.DeletePendingStopProposals(ctx, propsToExecute...)
```

Comment:
Same thing as above with the order of DeletePendingStopProposals and StopConsumerChain